pyiceberg/catalog/snowflake_catalog.py

```diff
@@ -0,0 +1,237 @@
+            parts = identifier.split(".")
+            if len(parts) == 1:
+                return cls(None, None, parts[0])
+            elif len(parts) == 2:
+                return cls(None, parts[0], parts[1])
+            elif len(parts) == 3:
+                return cls(parts[0], parts[1], parts[2])
+
+            raise ValueError(f"Invalid identifier: {identifier}")
+
+        @classmethod
+        def schema_from_string(cls, identifier: str) -> SnowflakeCatalog._SnowflakeIdentifier:
+            parts = identifier.split(".")
+            if len(parts) == 1:
+                return cls(None, parts[0], None)
+            elif len(parts) == 2:
+                return cls(parts[0], parts[1], None)
+
+            raise ValueError(f"Invalid identifier: {identifier}")
+
+        @property
+        def table_name(self) -> str:
+            return ".".join(self)
+
+        @property
+        def schema_name(self) -> str:
+            return ".".join(self)
+
+    def __init__(self, name: str, **properties: str):
+        super().__init__(name, **properties)
+
+        params = {
+            "user": properties["user"],
+            "account": properties["account"],
+        }
+
+        if "authenticator" in properties:
+            params["authenticator"] = properties["authenticator"]
+
+        if "password" in properties:
+            params["password"] = properties["password"]
+
+        if "private_key" in properties:
+            params["private_key"] = properties["private_key"]
+
+        self.connection = SnowflakeConnection(**params)
+
+    def load_table(self, identifier: Union[str, Identifier]) -> Table:
+        sf_identifier = SnowflakeCatalog._SnowflakeIdentifier.table_from_string(
+            identifier if isinstance(identifier, str) else ".".join(identifier)
+        )
+
+        metadata_query = "SELECT SYSTEM$GET_ICEBERG_TABLE_INFORMATION(%s) AS METADATA"
+
+        with self.connection.cursor(DictCursor) as cursor:
+            try:
+                cursor.execute(metadata_query, (sf_identifier.table_name,))
+                metadata = json.loads(cursor.fetchone()["METADATA"])["metadataLocation"]
+            except Exception as e:
+                raise NoSuchTableError(f"Table {sf_identifier.table_name} not found") from e
+
+        session = Session()
+        credentials = session.get_credentials()
+        current_credentials = credentials.get_frozen_credentials()
+
+        s3_props = {
+            S3_ACCESS_KEY_ID: current_credentials.access_key,
+            S3_SECRET_ACCESS_KEY: current_credentials.secret_key,
+            S3_SESSION_TOKEN: current_credentials.token,
+            S3_REGION: os.environ.get("AWS_REGION", "us-east-1"),
+        }
+
+        tbl = StaticTable.from_metadata(metadata, properties=s3_props)
+        tbl.identifier = tuple(identifier.split(".")) if isinstance(identifier, str) else identifier
+        tbl.catalog = self
+
+        return tbl
```

Comment:
It looks like it will be working only with AWS s3 ? Are you planing to add support for GCS, Azure?

Reply:
thanks for the callout. let me address this.

Reply:
i have added support for GCS now @questsul